### PR TITLE
[TD]Allow selection of Page when adding View

### PR DESCRIPTION
**Pull request by WandererFan**
If the user neglects to select a Page when adding a View, the add View request is ignored.  This PR adds a Page selection dialog that is invoked if the appropriate Page can not be determined automatically. 

Thank you for creating a pull request to contribute to FreeCAD! To ease integration, we ask you to conform to the following items. Pull requests which don't satisfy all the items below might be rejected. If you are in doubt with any of the items below, don't hesitate to ask for help in the [FreeCAD forum](https://forum.freecadweb.org/viewforum.php?f=10)!

- [ ]  Your pull request is confined strictly to a single module. That is, all the files changed by your pull request are either in `App`, `Base`, `Gui` or one of the `Mod` subfolders. If you need to make changes in several locations, make several pull requests and wait for the first one to be merged before submitting the next ones
- [ ]  In case your pull request does more than just fixing small bugs, make sure you discussed your ideas with other developers on the FreeCAD forum
- [ ]  Your branch is [rebased](https://git-scm.com/docs/git-rebase) on latest master `git pull --rebase upstream master`
- [ ]  All FreeCAD unit tests are confirmed to pass by running `./bin/FreeCAD --run-test 0`
- [ ]  All commit messages are [well-written](https://chris.beams.io/posts/git-commit/) ex: `Fixes typo in Draft Move command text`
- [ ]  Your pull request is well written and has a good description, and its title starts with the module name, ex: `Draft: Fixed typos`
- [ ]  Commit messages include `issue #<id>` or `fixes #<id>` where `<id>` is the [FreeCAD bug tracker issue number](https://freecadweb.org/wiki/tracker#GitHub_and_MantisBT) in case a particular commit solves or is related to an existing issue on the tracker. Ex: `Draft: fix typos - fixes #0004805`

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.20 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=56135).

---

### CODE REVIEW

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5309">pipeline status</a> for feature branch [PR_5309](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5309). Pipeline [439265241](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/439265241) was triggered at [ed0c57f](https://github.com/FreeCAD/FreeCAD/pull/5309/commits/ed0c57f69ddb690d2cc090592c49df4593e366bb). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

File: src/Mod/TechDraw/Gui/DrawGuiUtil.cpp [hunk 1]
```diff
@@ -127,17 +130,37 @@ TechDraw::DrawPage* DrawGuiUtil::findPage(Gui::Command* cmd)
                 page = qp->getDrawPage();
             } else {
                 // no active page
-                QMessageBox::warning(Gui::getMainWindow(), QObject::tr("Which page?"),
-                                     QObject::tr("Can not determine correct page."));
+                for(auto obj: selPages) {
```
Comment (donovaly) [inline src/Mod/TechDraw/Gui/DrawGuiUtil.cpp:133]: Just a trifle: we have no official style rules but try to use uniformly spaces behind for, if, else etc. In the file this style is already used.

**Comment by donovaly** [comment]
Many thanks. I only commented on a very minor style issue since I first can compile next week to give useful feedback.

**Comment by freecadci** [comment]
<a href="https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5309">pipeline status</a> for feature branch [PR_5309](https://gitlab.com/freecad/FreeCAD-CI/-/commits/PR_5309). Pipeline [439855198](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines/439855198) was triggered at [91725fb](https://github.com/FreeCAD/FreeCAD/pull/5309/commits/91725fb284848ab4b600239ae4678020890ef41d). All CI [branches](https://gitlab.com/freecad/FreeCAD-CI/-/branches/all) and [pipelines](https://gitlab.com/freecad/FreeCAD-CI/-/pipelines?scope=branches).

**Pull request merged**